### DISCUSSION TITLE
Add `grid-template-rows` with `0fr` and `1fr` for height animation from 0 to auto

### DISCUSSION THREAD

**Discussion by wm-simon** [tailwindlabs/tailwindcss#11186] [Ideas]
Hi there.,

as described in this article, the height of an element can easily be animated from 0 to auto using the properties `grid-template-rows: 0fr` and `grid-template-rows: 1fr`. It would be good if these properties were added as a standard.
https://keithjgrant.com/posts/2023/04/transitioning-to-height-auto/

**Comment by sennett** [comment]
You can make your own by extending [`tailwind.config.js`](https://tailwindcss.com/docs/theme) like this:

```javascript
export default {
  theme: {
    extend: {
      gridTemplateRows: {
        "animate-height-open": "1fr",
        "animate-height-closed": "0fr",
      },
    },
  },
};
```

Then consume in components:

```jsx
<MyAnimatedHeightWithGrid classNames="grid  grid-rows-animate-height-open  transition-all  duration-700  ease-in-out" />

```

**Comment by czepiec** [comment]
You can also use also `grid-cols-[0fr]` or `grid-rows-[0fr]` classes like:

```html
<div id="myElement" class="grid grid-cols-[0fr] duration-500">
  <div class="overflow-hidden">
    ...
  </div>
</div>
```

and `onclick="document.getElementById('myElement').classList.add('grid-cols-[1fr]');"` (add or remove) somewhere.
